drivers/pds/targetcluster/targetcluster.go

```diff
@@ -85,9 +85,13 @@ func (tc *TargetCluster) GetDeploymentTargetID(clusterID, tenantID string) (stri
 			targetClusterStatus = targetClusters[i].GetStatus()
 		}
 	}
-	if targetClusterStatus != "healthy" {
-		return "", fmt.Errorf("target Cluster is not in healthy state due to error : %v", err)
-	}
+	err = wait.Poll(DefaultRetryInterval, 5*time.Minute, func() (bool, error) {
+		if targetClusterStatus != "healthy" {
```

Comment:
targetClusterStatus will not get updated, unless you call 
`targetClusters, err := components.DeploymentTarget.ListDeploymentTargetsBelongsToTenant(tenantID)` this inside the polling method.

Reply:
```
if targetClusterStatus == "healthy" {
   return true, nil
}
```